Makefile

```diff
@@ -1,37 +1,71 @@
-ENV=prod
-REGION=.learnosity.com
-# Data API
-VER=v1
+ifndef LRN_SDK_NO_DOCKER
+CMP_PROJECT = lrn-sdk-java
+CMP = docker-compose -f lrn-dev/docker-compose.yml --env-file lrn-dev/env --project-name $(CMP_PROJECT)
+CMP_RUN = $(CMP) run --rm java
+CMP_BUILD = $(CMP) build
+CMP_CLEAN = docker image ls --filter reference=$(CMP_PROJECT)\* -q | xargs docker rmi
+CMP_RM_VOLUME = docker volume rm $(CMP_PROJECT)_repo
+endif
 
-all: build test
+PROJECT_VERSION_CMD = mvn org.apache.maven.plugins:maven-help-plugin:3.2.0:evaluate -Dexpression=project.version -q -DforceStdout
+PKG_VER = v$(shell $(CMP_RUN) $(PROJECT_VERSION_CMD))
+GIT_TAG = $(shell git describe --tags)
+VERSION_MISMATCH = For a release build, the package version number $(PKG_VER) must match the git tag $(GIT_TAG).
 
-dist: prodbuild
+# Data API settings
 
-test: test-unit test-integration-env
+DAPI_ENV = prod
+DAPI_REGION = .learnosity.com
+DAPI_VER = v1
 
-clean: build-clean
+# Dev cycle targets
 
-dist-check-version: PKG_VER=v$(shell sed -n "s/^.*<version>\([^']\+\)<\/version>.*$$/\1/p;T;q" pom.xml)
-dist-check-version: GIT_TAG=$(shell git describe --tags)
-dist-check-version:
-ifeq ('$(shell echo $(GIT_TAG) | grep -qw "$(PKG_VER)")', '')
-	$(error Version number $(PKG_VER) in setup.py does not match git tag $(GIT_TAG))
-endif
+all: build test
 
-prodbuild: dist-check-version build test
-	mvn package
-devbuild: build
 build:
-	mvn compile
+	$(CMP_RUN) mvn compile
+
+test: test-unit test-integration-env
 
 test-unit:
-	mvn -q test
+	$(CMP_RUN) mvn test
+
 test-integration-env:
-	mvn -q integration-test
+	ENV=$(DAPI_ENV) REGION=$(DAPI_REGION) VER=$(DAPI_VER) $(CMP_RUN) mvn integration-test
+
+clean:
+	$(CMP_RUN) mvn clean
+
+prodbuild: version-check build test
+	$(CMP_RUN) mvn package
+
+version-check-message:
+	@echo Checking git and project versions ...
+
+version-check: version-check-message
+	@echo $(GIT_TAG) | grep -qw "$(PKG_VER)" || (echo $(VERSION_MISMATCH); exit 1)
+
+# Some target aliases
+
+dist: prodbuild
+
+devbuild: build
+
+build-clean: clean
```

Comment:
We don't need build-clean any more - that's a relic, it was just left in for legacy builds, but we have updated that all now

Reply:
OK, I'll remove it.